src/lib-components/DateFilter.vue

```diff
@@ -0,0 +1,607 @@
+    todayBtnActive.value = true
+  }
+}
+// Clear date selection
+function clearDate() {
+  datepicker.value?.updateInternalModelValue(null)
+  todayBtnActive.value = false
+}
+// When selection changes, determine is-selecting boolean for range selection styles
+function handleInternalSelection(selectedDate: Date | Date[] | null) {
+  if ((selectedDate && 'length' in selectedDate) && selectedDate.length.valueOf() === 1)
+    isSelecting.value = true
+  else
+    isSelecting.value = false
+}
+// Deselect today button when range selection starts
+function clearTodayBtn() {
+  todayBtnActive.value = false
+}
+// Toggle the arrow on the datepicker between open and closed
+function toggleArrow() {
+  isOpen.value = !isOpen.value
+}
+
+// EXPOSED METHODS - Parent component can use these methods to open and close the datepicker if input is hidden
+// Open datepicker
+function openDatepicker() {
+  datepicker.value?.openMenu()
+}
+// Close datepicker
+function closeDatepicker() {
+  datepicker.value?.closeMenu()
+}
+defineExpose({
+  openDatepicker,
+  closeDatepicker,
+})
+
+// COMPUTED VALUES
+// Determine if the window size is mobile for conditional rendering
+const isMobile = computed(() => globalStore.winWidth <= 750)
+// Format the selected date(s) into consistent object
+const formattedDateSelection = computed(() => {
+  // range selected
+  if (date.value && 'length' in date.value) {
+    return {
+      startDate: date.value[0],
+      endDate: date.value[1],
+    }
+  }
+  // single date selected
+  else if (date.value) {
+    return {
+      startDate: date.value,
+      endDate: null,
+    }
+  }
+  // nothing selected
+  return {
+    startDate: null,
+    endDate: null,
+  }
+})
+
+// ASYNC COMPONENTS
+const ButtonLink = defineAsyncComponent(() =>
+  import('@/lib-components/ButtonLink.vue'))
+
+// Watch date and emit to parent component
+watch(date, async (newDate, oldDate) => {
+  if (newDate !== oldDate) {
+    // change seperator for single or range select
+    if ((newDate && 'length' in newDate) && newDate[1] === null) {
+      textConfig.value = {
+        rangeSeparator: ' ',
+      }
+    }
+    else {
+      textConfig.value = {
+        rangeSeparator: ' — ',
+      }
+    }
+    // eslint-disable-next-line vue/custom-event-name-casing
+    emit('input-selected', formattedDateSelection.value)
+  }
+})
+// Watch window size and update windowSize ref
+/* window.addEventListener('resize', () => {
+  windowSize.value = window.innerWidth
+}) */
+</script>
+
+<template>
+  <div class="date-filter-container">
+    <VueDatePicker
+      ref="datepicker" v-model="date" :config="vue3datepickerConfig" :range="!isMobile" :week-start="0"
+      month-name-format="long" :enable-time-picker="false" :auto-position="false" :auto-apply="true"
+      :text-input="textConfig" no-today :inline="hideInput" class="date-filter"
+      :class="[{ 'is-selecting': isSelecting }]" :placeholder="isMobile ? 'Select a date' : 'All upcoming'"
```

Comment:
move any logic into computed method named datePickerClass = computed(() => [{ 'is-selecting': isSelecting }] )